### PR TITLE
Added Azure.Defender.CosmosDb

### LINKED ISSUE #2204: Check Defender for Azure Cosmos DB provided by Microsoft Defender for Cloud is using Standard plan
### Existing rule

_No response_

### Suggested rule

Cosmos DBs should use Microsoft Defender for Azure Cosmos DB which is a feature of Microsoft Defender for Cloud.

Microsoft Defender for Azure Cosmos DB detects potential SQL injections, known bad actors based on Microsoft Threat Intelligence, suspicious access patterns, and potential exploitation of databases through compromised identities, or malicious insiders.

When enabling the Defender pricing plan for Cosmos DB it is automatically enabled for all Azure Cosmos DB for NoSQL accounts within the subscription.

This is done by enabling the Cosmos DB pricing plan (section) under the parent databases pricing plan in MDfC.

With infrastructure as code, PowerShell this or CLI this is done by is it done by setting a Standard `CosmosDbs` pricing plan.

Related to #1632 

### Pillar

Security

### Additional context

https://learn.microsoft.com/azure/defender-for-cloud/concept-defender-for-cosmos
https://learn.microsoft.com/azure/defender-for-cloud/defender-for-databases-enable-cosmos-protections
https://learn.microsoft.com/azure/templates/microsoft.security/advancedthreatprotectionsettings

### PR DESCRIPTION
**Pull request by BenjaminEngeset**
## PR Summary

Fixes #2204
Fixes a topic within #1632 

Added Azure.Defender.CosmosDb rule.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by BenjaminEngeset** [comment]
@BernieWhite Let me know :)

File: docs/en/rules/Azure.Defender.CosmosDb.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+# Set Microsoft Defender for Cosmos DB to the Standard tier
+
+## SYNOPSIS
+
+Enable Microsoft Defender for Azure Cosmos DB.
+
+## DESCRIPTION
+
+Microsoft Defender for Azure Cosmos DB provides additional security for Azure Cosmos DB accounts.
```
Comment (BernieWhite) [inline docs/en/rules/Azure.Defender.CosmosDb.md:17]:
```suggestion
Microsoft Defender for Azure Cosmos DB provides additional security insight for Azure Cosmos DB accounts.
```

File: docs/en/rules/Azure.Defender.CosmosDb.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+Protection is provided by analyzing onboarded Cosmos DB accounts for unusual and potentially harmful attempts to access or exploit the accounts.
+Which allows Microsoft Defender for Cloud to produce security alerts that are triggered when anomalies in activity occur.
+
+Security alerts for onboarded Cosmos DB accounts shows up in Defender for Cloud with details of the suspicious activity and recommendations on how to investigate and remediate the threats.
+
+Microsoft Defender for Cosmos DB can be enabled at the subscription level and by doing so ensures all Cosmos DB accounts in the subscription will be protected, including future ones.
+
+## RECOMMENDATION
+
+Consider using Microsoft Defender for Azure Cosmos DB to provide additional security for Azure Cosmos DB accounts.
```
Comment (BernieWhite) [inline docs/en/rules/Azure.Defender.CosmosDb.md:28]:
```suggestion
Consider using Microsoft Defender for Azure Cosmos DB to provide additional security insights for Azure Cosmos DB accounts.
```

File: docs/en/rules/Azure.Defender.CosmosDb.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+For example:
+
+```powershell
+Set-AzSecurityPricing -Name 'CosmosDbs' -PricingTier 'Standard'
+```
+
+## NOTES
+
+Microsoft Defender for Azure Cosmos DB is currently available only for the API for NoSQL.
```
Comment (BernieWhite) [inline docs/en/rules/Azure.Defender.CosmosDb.md:94]:
```suggestion
Microsoft Defender for Azure Cosmos DB is currently available only for the NoSQL API.
```

File: src/PSRule.Rules.Azure/rules/Azure.Defender.Rule.yaml [hunk 4]
```diff
@@ -266,4 +266,28 @@ spec:
...
+kind: Rule
+metadata:
+  name: Azure.Defender.CosmosDb
+  ref: AZR-000379
+  tags:
+    release: GA
+    ruleSet: 2023_06
+    Azure.WAF/pillar: Security
+  labels:
+    Azure.MCSB.v1/control: LT-1
```
Comment (BernieWhite) [inline src/PSRule.Rules.Azure/rules/Azure.Defender.Rule.yaml:281]:
```suggestion
    Azure.MCSB.v1/control: [ 'DP-2', 'LT-1' ]
```

File: docs/en/rules/Azure.Defender.CosmosDb.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+## LINKS
+
+- [Security operations in Azure](https://learn.microsoft.com/azure/architecture/framework/security/monitor-security-operations)
+- [What is Microsoft Defender for Cloud?](https://learn.microsoft.com/azure/defender-for-cloud/defender-for-cloud-introduction)
+- [Overview of Microsoft Defender for Azure Cosmos DB](https://learn.microsoft.com/azure/defender-for-cloud/concept-defender-for-cosmos)
+- [Enable Microsoft Defender for Azure Cosmos DB](https://learn.microsoft.com/azure/defender-for-cloud/defender-for-databases-enable-cosmos-protections)
+- [Quickstart: Enable enhanced security features](https://learn.microsoft.com/azure/defender-for-cloud/enable-enhanced-security)
+- [Azure security baseline for Azure Cosmos DB](https://learn.microsoft.com/security/benchmark/azure/baselines/azure-cosmos-db-security-baseline)
+- [LT-1: Enable threat detection capabilities](https://learn.microsoft.com/security/benchmark/azure/baselines/azure-cosmos-db-security-baseline#lt-1-enable-threat-detection-capabilities)
```
Comment (BernieWhite) [inline docs/en/rules/Azure.Defender.CosmosDb.md:104]:
```suggestion
- [DP-2: Monitor anomalies and threats targeting sensitive data](https://learn.microsoft.com/security/benchmark/azure/baselines/azure-cosmos-db-security-baseline#dp-2-monitor-anomalies-and-threats-targeting-sensitive-data)
- [LT-1: Enable threat detection capabilities](https://learn.microsoft.com/security/benchmark/azure/baselines/azure-cosmos-db-security-baseline#lt-1-enable-threat-detection-capabilities)
```

**Review by BernieWhite — CHANGES_REQUESTED** [review]
Thanks @BenjaminEngeset. Great work!

I've added a few suggestions, otherwise all good to merge.

**Review by BernieWhite — CHANGES_REQUESTED** [review]
Thanks @BenjaminEngeset. Great work!

I've added a few suggestions, otherwise all good to merge.

**Comment by BenjaminEngeset** [comment]
@BernieWhite Great and thanks for the feedback. I have updated the rule now.

**Review by BernieWhite — APPROVED** [review]
Thanks @BenjaminEngeset. All good to merge.

**Pull request merged**